CHERI-UPDATING.md

```diff
@@ -79,6 +79,152 @@ Each merge from upstream FreeBSD to [dev] is accompanied by a tag of the form
 We typically merge to [master] from [dev] at stable points at least a week
 apart.
 
+### The merge process
+CheriBSD is extremely diverged from FreeBSD with nearly two
+thousand changed files. As such, there are many opportunities for
+a merge to conflict and worse for a merge to not conflict, but
+be broken. It is critical that we be able to bisect the merged
+changes. To support this we merge changes one commit at a time
+from the _first-parent_ history of FreeBSD's `main` using the
+[mergify](https://github.com/brooksdavis/mergify) script. This makes it
+straightforward to discover which change caused a problem.
+
+This section describes the merge process in detail.
+
+#### Setup
+The merger must add an appropriate FreeBSD remote to their setup. This
+example uses the official FreeBSD mirror, a mirror on GitHub or the like
+will also work fine.
+```
+$ git remote add upstream https://git.FreeBSD.org/src.git
+```
+You will also want to enable `git revere` to simplify handling the same
+conflicting merges over again. To do so globally, run:
+```
+$ git config --global rerere.enabled true
+```
+Disable rename detection. This sometimes leads to awkward merges but
+rename detection takes **forever** on the CheriBSD repo.
+
+#### Identifying the merge point
+Update the upstream remote if required:
+```
+$ git fetch upstream
+```
+Then identify the head of the set of commits to merge. Because FreeBSD
+does not enforce date linearity this is a bit annoying. The default
+`Date:` will generally be somewhere in the past relative to the actual
+push unless the committer reset the date immediately before with
+`git rebase --ignore-date` or `git commit --amend --date=now`. As
+such we want to look at the commit date which is a better (but still
+sub-optimal) proxy for the time the change was pushed to the FreeBSD
+tree. We currently do this by examining the log and looking for the
+first `CommitDate: ` line mentioning the day we wish to merge. To see
+only eligible commits and to cause the commit date to be displayed we
+use:
+```
+$ git log --first-parent --format=fuller upstream/main
+```
+We can then use the pager's search function to search for the
+appropriate date (e.g.. `Commit.*Fri Feb 12`). It's a good idea to check
+a few commits up or down to make sure the commit isn't at an obviously
+bad point. Then we create a tag for the commit:
+```
+$ git tag -m "FreeBSD main DD Month YYYY" freebsd-main-<YYYYMMDD> <hash>
+```
+
+#### Performing the merge
+First, create a merge branch from an update to date `dev` checkout:
+```
+$ git checkout -b merge-freebsd-<YYYYMMDD>
```

Comment:
I think you can use
```suggestion
$ git checkout -b merge-freebsd-<YYYYMMDD> dev
```
to help ensure that you start from the right place, regardless of the current checkout?